src/java.base/share/classes/java/lang/foreign/Arena.java

```diff
@@ -0,0 +1,159 @@
+ * can be {@linkplain SegmentScope#isAccessibleBy(Thread) accessed} by any thread. This might be useful when
+ * multiple threads need to access the same memory segment concurrently (e.g. in the case of parallel processing).
+ * Moreover, a shared arena {@linkplain #isCloseableBy(Thread) can be closed} by any thread.
+ *
+ * @since 20
+ */
+@PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
+public interface Arena extends SegmentAllocator, AutoCloseable {
+
+    /**
+     * Returns a native memory segment with the given size (in bytes) and alignment constraint (in bytes).
+     * The returned segment is associated with the arena scope.
+     * The segment's {@link MemorySegment#address() address} is the starting address of the
+     * allocated off-heap memory region backing the segment, and the address is
+     * aligned according the provided alignment constraint.
+     *
+     * @implSpec
+     * The default implementation of this method is equivalent to the following code:
+     * {@snippet lang = java:
+     * MemorySegment.allocateNative(bytesSize, byteAlignment, scope());
+     *}
+     * More generally implementations of this method must return a native segment featuring the requested size,
+     * and that is compatible with the provided alignment constraint. Furthermore, for any two segments
+     * {@code S1, S2} returned by this method, the following invariant must hold:
+     *
+     * {@snippet lang = java:
+     * S1.overlappingSlice(S2).isEmpty() == true
+     *}
+     *
+     * @param byteSize the size (in bytes) of the off-heap memory block backing the native memory segment.
+     * @param byteAlignment the alignment constraint (in bytes) of the off-heap region of memory backing the native memory segment.
+     * @return a new native memory segment.
+     * @throws IllegalArgumentException if {@code bytesSize < 0}, {@code alignmentBytes <= 0}, or if {@code alignmentBytes}
+     * is not a power of 2.
+     * @throws IllegalStateException if the arena has already been {@linkplain #close() closed}.
+     * @throws WrongThreadException if this method is called from a thread {@code T},
+     * such that {@code scope().isAccessibleBy(T) == false}.
+     * @see MemorySegment#allocateNative(long, long, SegmentScope)
+     */
+    @Override
+    default MemorySegment allocate(long byteSize, long byteAlignment) {
+        return MemorySegment.allocateNative(byteSize, byteAlignment, scope());
+    }
+
+    /**
+     * {@return the arena scope}
+     */
+    SegmentScope scope();
+
+    /**
+     * Closes this arena. If this method completes normally, the arena scope is no longer {@linkplain SegmentScope#isAlive() alive},
+     * and all the memory segments associated with it can no longer be accessed. Furthermore, any off-heap region of memory backing the
+     * segments associated with that scope are also released.
+     * @throws IllegalStateException if the arena has already been {@linkplain #close() closed}.
```

Comment:
JavaDoc was pointing to itself.
```suggestion
     * @throws IllegalStateException if the arena has already been closed.
```